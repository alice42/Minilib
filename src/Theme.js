export const theme = {
  backgroundColor: '#ECEFF1',
  onBackground: '#000000',
  surfaceColor: '#FFFFFF',
  surfaceColorHover: '#F5F5F5',
  surfaceColorActive: '#E0E0E0',
  onSurface: '#000000',
  primaryColor: '#0277BD',
  primaryColorHover: '#0288D1',
  primaryColorActive: '#01579B',
  onPrimary: '#FFFFFF',
  secondaryColor: '#00897B',
  secondaryColorHover: '#00695C',
  secondaryColorActive: '#004D40',
  onSecondary: '#FFFFFF',
  tertiaryColor: '#EF6C00',
  tertiaryColorHover: '#FB8C00',
  tertiaryColorActive: '#E65100',
  onTertiary: '#FFFFFF',
  fontPrimary: {
    regular: 'Lato,sans-serif',
    bold: 'Lato,sans-serif',
    light: 'Lato,sans-serif'
  },
  fontSecondary: {
    regular: 'arial,sans-serif',
    bold: 'arial,sans-serif',
    light: 'arial,sans-serif'
  },
  fontTertiary: {
    regular: 'Georgia,serif',
    bold: 'Georgia,serif',
    light: 'Georgia,serif'
  },
  font48: '4rem',
  font43: '3.6rem',
  font38: '3.2rem',
  font34: '2.9rem',
  font30: '2.5rem',
  font29: '2.4rem',
  font28: '2.3rem',
  font26: '2.1rem',
  font24: '2rem',
  font23: '1.9rem',
  font22: '1.85rem',
  font20: '1.65rem',
  font19: '1.6rem',
  font18: '1.5rem',
  font17: '1.4rem',
  font16: '1.35rem',
  font15: '1.25rem',
  font14: '1.2rem',
  font13: '1.1rem',
  font12: '1rem',
  font11: '0.95rem',
  font10: '0.9rem',
  font9: '0.75rem',
  font8: '0.7rem',
  font6: '0.5rem',
  overlayOpacity: 0.7,
  borderRadius: '5px',
  border: 'solid 5px #000000'
}
