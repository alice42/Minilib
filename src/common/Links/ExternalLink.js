import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ExternalLinkStyled } from './ExternalLinkStyled'

export default class ExternalLink extends Component {
  static propTypes = {
    /** redirection URL */
    href: PropTypes.string,
    /** content text */
    label: PropTypes.string,
    /** override CSS */
    customStyle: PropTypes.object
  }

  static defaultProps = {
    href: null,
    label: null,
    customStyle: {}
  };

  render = () => <ExternalLinkStyled {...this.props}> {this.props.label || this.props.children} </ExternalLinkStyled>
}
