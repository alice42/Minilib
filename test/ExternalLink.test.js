import React from 'react'
import ExternalLink from '../src/common/Links/ExternalLink'
import 'jest-styled-components'
import { renderCreateWithTheme } from './TestUtils'
import { theme } from '../src/Theme.js'
import { shallow } from 'enzyme'

const url = 'https://www.veepee.com/fr-fr/'

test('Basic link', () => {
  const component = renderCreateWithTheme(
    <ExternalLink
      href={url}
      label='Pretty basic button'
    />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-family', theme.fontPrimary.regular)
  expect(tree.props.href).toBe(url)
})

const redHover = {
  ':hover': {
    'color': 'red'
  }
}

test('Basic link with custom css (styled-component)', () => {
  const component = renderCreateWithTheme(
    <ExternalLink
      href={url}
      label='Pretty custom link'
      customStyle={redHover}
    />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-family', theme.fontPrimary.regular)
  expect(tree).toHaveStyleRule(redHover)
  expect(tree.props.href).toBe(url)
})

const buttonStyle = {
  'background-color': theme.primaryColor,
  'display': 'flex',
  'justify-content': 'center',
  'align-items': 'center',
  'height': '40px',
  'width': '10%',
  'font-family': theme.fontPrimary.bold,
  'text-transform': 'uppercase',
  'color': 'white',
  'text-decoration': 'none',
  'border-radius': '10px',
  ':hover': {
    'color': theme.primaryColor,
    'background-color': 'white',
    'border': `solid 1px ${theme.primaryColor}`
  }
}

test('Link as button', () => {
  const component = renderCreateWithTheme(
    <ExternalLink
      href={url}
      label='Pretty custom button'
      customStyle={buttonStyle}
    />
  )
  let tree = component.toJSON()

  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('font-family', theme.fontPrimary.bold)
  expect(tree).toHaveStyleRule(buttonStyle)
  expect(tree.props.href).toBe(url)
})

test('Link with children', () => {
  const component = shallow(
    <ExternalLink href={url} customStyle={buttonStyle}>
      <div id='div'>Click Here</div>
    </ExternalLink>)

  expect(component).toMatchSnapshot()
  expect(component.find('#div').text()).toBe('Click Here')
  expect(component.props().href).toBe(url)
})
