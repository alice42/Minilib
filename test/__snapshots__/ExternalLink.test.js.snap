// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic link 1`] = `
.c0 {
  font-family: Lato,sans-serif;
}

<a
  className="c0"
  href="https://www.veepee.com/fr-fr/"
  label="Pretty basic button"
>
   
  Pretty basic button
   
</a>
`;

exports[`Basic link with custom css (styled-component) 1`] = `
.c0 {
  font-family: Lato,sans-serif;
}

.c0:hover {
  color: red;
}

<a
  className="c0"
  href="https://www.veepee.com/fr-fr/"
  label="Pretty custom link"
>
   
  Pretty custom link
   
</a>
`;

exports[`Link as button 1`] = `
.c0 {
  font-family: Lato,sans-serif;
  background-color: #0277BD;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 40px;
  width: 10%;
  font-family: Lato,sans-serif;
  text-transform: uppercase;
  color: white;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-radius: 10px;
}

.c0:hover {
  color: #0277BD;
  background-color: white;
  border: solid 1px #0277BD;
}

<a
  className="c0"
  href="https://www.veepee.com/fr-fr/"
  label="Pretty custom button"
>
   
  Pretty custom button
   
</a>
`;

exports[`Link with children 1`] = `
<styled.a
  customStyle={
    Object {
      ":hover": Object {
        "background-color": "white",
        "border": "solid 1px #0277BD",
        "color": "#0277BD",
      },
      "align-items": "center",
      "background-color": "#0277BD",
      "border-radius": "10px",
      "color": "white",
      "display": "flex",
      "font-family": "Lato,sans-serif",
      "height": "40px",
      "justify-content": "center",
      "text-decoration": "none",
      "text-transform": "uppercase",
      "width": "10%",
    }
  }
  href="https://www.veepee.com/fr-fr/"
  label={null}
>
   
  <div
    id="div"
  >
    Click Here
  </div>
   
</styled.a>
`;
